### PR TITLE
create curl pod as fixture

### PR DESCRIPTION
**Pull request by beckmani**

### CODE REVIEW

File: tests/k8s/curl.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+spec:
+  containers:
+  - name: curl
+    image: byrnedo/alpine-curl
+    command:
+      - "sh"
+      - "-c"
+      - >
+        while true; do
+          sleep 3600;
```
Comment (antonlerner) [inline tests/k8s/curl.yml:16]: why is this needed?

File: tests/test_bs.py [hunk 2]
```diff
@@ -381,6 +382,25 @@ def save_log_on_exit(request):
         (out, err) = p.communicate()
 
 
+@pytest.fixture()
+def add_curl(request, setup_bootstrap):
```
Comment (y0sher) [inline tests/test_bs.py:386]: I don't get why bootstrap is required here?

File: tests/test_bs.py [hunk 3]
```diff
@@ -417,7 +437,7 @@ def test_add_client(add_client):
     assert len(hits) == 1, "Could not find new Client bootstrap message"
 
 
-def test_gossip(setup_clients):
+def test_gossip(setup_clients, add_curl):
```
Comment (y0sher) [inline tests/test_bs.py:440]: isn't it better to add it as a precondition to a more basic fixture like the the one that sets the namespace?

File: Makefile [hunk 4]
```diff
@@ -166,8 +168,12 @@ endif
              -e GOOGLE_APPLICATION_CREDENTIALS=./spacemesh.json \
              -e CLIENT_DOCKER_IMAGE="spacemeshos/$(DOCKER_IMAGE_REPO):$(BRANCH)" \
              -it go-spacemesh-python pytest -s sync/test_sync.py --tc-file=sync/config.yaml --tc-format=yaml
-
-
+else
```
Comment (antonlerner) [inline Makefile]: this can be refactored without duplication of the entire command,
use ifndef TEST -> delimiter='::' else delimiter=''
and then the last line should look like this
-it go-spacemesh-python pytest -s test_bs.py$(Delimiter)$(TEST) --tc-file=config.yaml --tc-format=yaml

File: tests/test_bs.py (hunk 2 above)
Comment (antonlerner) [inline tests/test_bs.py:386]: we have too many fixtures at the begining of each test, consider making this in higher level so it won't need to be specified each time

File: tests/k8s/curl.yml (hunk 1 above)
Comment (beckmani) [reply tests/k8s/curl.yml:16]: To keep the container running by docker engine

File: tests/test_bs.py (hunk 2 above)
Comment (beckmani) [reply tests/test_bs.py:386]: setup_bootstrap holds the deployment_id, this is the string common to all the deployed parts (bootstrap, client, poet, oracle etc.)

File: tests/test_bs.py (hunk 3 above)
Comment (beckmani) [reply tests/test_bs.py:440]: I added it here because there are few tests that don't need the curl.

File: Makefile (hunk 4 above)
Comment (beckmani) [reply Makefile]: ok.

File: tests/test_bs.py (hunk 2 above)
Comment (beckmani) [reply tests/test_bs.py:386]: I'm not sure I understand this comment.
Fixtures have order. each fixture depends on other fixtures.
To deploy full network you need to call only 1 fixture - setup_clients --> this fixture will call other fixtures as needed.

Comment (y0sher) [reply tests/test_bs.py:386]: yes but we don't use it in the code below. ( you check that pods are there but that's not needed ).  we only use the namespace from testconfig

**Review by antonlerner — APPROVED** [review]
